pretyping/typeclasses.ml

```diff
@@ -130,22 +138,19 @@ let rec is_class_type evd c =
     match EConstr.kind evd c with
     | Prod (_, _, t) -> is_class_type evd t
     | Cast (t, _, _) -> is_class_type evd t
+    | Proj (p, c) -> GlobRef.(Map.mem (ConstRef (Projection.constant p))) !classes
```

Comment:
This is adding a new dependency to the legacy constant binding mechanism. I know we're a bit in a hurry w.r.t. the release, but instead of doing that we should have a type of global references including projections. I can validate the PR as is, but if you have time, could you do that the right way from the start?

Reply:
Hmm, I don't really have time right now, but indeed that should happen next. Note that we already had code using `Projection.constant` here, so it's not really worse than before.